### PR TITLE
ECL-483: Interests On Overdue Payment

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyaccount/services/FinancialDataService.scala [hunk 1]
```diff
@@ -125,7 +124,26 @@ class FinancialDataService @Inject() (
           }
       }
 
-    FinancialViewDetails(outstandingPayments = outstandingPayments, paymentHistory = paymentsHistory)
+    val accruingInterestOutstandingPayments = documentDetails
+      .collect(filterItemsThatHaveAccruingInterest)
+      .filter(document => document.interestAccruingAmount.nonEmpty)
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyaccount/services/FinancialDataService.scala:129]: Isn't this redundant if **filterItemsThatHaveAccruingInterest** already has **filterItemsThatHaveAccruingInterestAmount**?

**Review by TheDanielDoyle — CHANGES_REQUESTED** [review]
See comment

File: app/uk/gov/hmrc/economiccrimelevyaccount/services/FinancialDataService.scala (hunk 1 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyaccount/services/FinancialDataService.scala]: Oh, correct correct, left it in during the refactoring. Thank you, it's removed now.

**Review by liandra-giacomine — CHANGES_REQUESTED** [review]
need unit tests for these changes

**Review by liandra-giacomine — APPROVED** [review]

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**